### PR TITLE
Bump MSRV to Rust 1.80

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Prerequisit for https://github.com/astral-sh/ruff/pull/13654

UV's MSRV is 1.81

## Test Plan

cargo test

### CODE REVIEW

File: crates/ruff_linter/Cargo.toml [hunk 1]
```diff
@@ -71,6 +70,7 @@ unicode-width = { workspace = true }
 unicode_names2 = { workspace = true }
 unicode-normalization = { workspace = true }
 url = { workspace = true }
+once_cell = "1.20.2"
```
Comment (AlexWaygood) [inline crates/ruff_linter/Cargo.toml:73]: Did you mean to get rid of this dependency? It looks like you just moved it in this file

**Review by AlexWaygood — APPROVED** [review]
I think there may be other modernisations we can do as a result of this. For example, I know that we can use non-deprecated `cargo` commands in `build.rs` files with this MSRV bump: all of these can be `cargo::` rather than `cargo:`:

```
crates/red_knot_python_semantic/build.rs:    println!("cargo::rerun-if-changed=resources/mdtest");
crates/red_knot_vendored/build.rs:    println!("cargo:rerun-if-changed={TYPESHED_SOURCE_DIR}");
crates/ruff/build.rs:    println!("cargo:rustc-env=RUST_HOST_TARGET={target}");
crates/ruff/build.rs:        "cargo:rerun-if-changed={}",
crates/ruff/build.rs:                "cargo:rerun-if-changed={}",
crates/ruff/build.rs:    println!("cargo:rustc-env=RUFF_COMMIT_HASH={}", next());
crates/ruff/build.rs:    println!("cargo:rustc-env=RUFF_COMMIT_SHORT_HASH={}", next());
crates/ruff/build.rs:    println!("cargo:rustc-env=RUFF_COMMIT_DATE={}", next());
crates/ruff/build.rs:            "cargo:rustc-env=RUFF_LAST_TAG={}",
crates/ruff/build.rs:            "cargo:rustc-env=RUFF_LAST_TAG_DISTANCE={}",
```

But I think it's fine to do that kind of thing as a followup

File: crates/ruff_linter/Cargo.toml (hunk 1 above)
Comment (MichaReiser) [reply crates/ruff_linter/Cargo.toml:73]: lol, Rust Rover must have inserted the dependency again :D

Comment (MichaReiser) [reply crates/ruff_linter/Cargo.toml:73]: That also explains why the error count in ruff_linter suddenly dropped from 20 to 0

**Comment by MichaReiser** [comment]
I didn't know about the `cargo` change but I agree that we can do some of them as follow ups. I did go through the announcement blog posts to learn about obvious things (like the `once_cell`) that we can use now.

**Pull request merged**